source/common/http/status.cc

```diff
@@ -0,0 +1,123 @@
+#include "common/http/status.h"
+
+#include "absl/strings/str_cat.h"
+
+namespace Envoy {
+namespace Http {
+
+namespace {
+
+constexpr absl::string_view EnvoyPayloadUrl = "Envoy";
+
+absl::string_view statusCodeToString(StatusCode code) {
+  switch (code) {
+  case StatusCode::Ok:
+    return absl::OkStatus().ToString();
+  case StatusCode::CodecProtocolError:
+    return "CodecProtocolError";
+  case StatusCode::BufferFloodError:
+    return "BufferFloodError";
+  case StatusCode::PrematureResponseError:
+    return "PrematureResponseError";
+  case StatusCode::CodecClientError:
+    return "CodecClientError";
+  }
```

Comment:
we're getting compilation issues on Windows as this helper function does not return a value in all cases, we're going to make a PR to fix this and add a `RELEASE_ASSERT`  in the `default` case (and change all all the below `ASSERT`s to `RELEASE_ASSERT`s), if an exception is more appropriate, please let us know

Reply:
The documentation of `getStatusCode` mentions a `RELEASE_ASSERT` but is using `ASSERT`

Reply:
Disregard the part about `ASSERT` -> `RELEASE_ASSERT` as I see the comments now asking to do the opposite